06_puppy/kasunfdo/mapstore.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+	"fmt"
+)
+
+// MapStore is map based implementation of Storer interface
+type MapStore struct {
+	store  map[uint64]Puppy
+	nextID uint64
+}
+
+// NewMapStore creates a new MapStore
+func NewMapStore() *MapStore {
+	return &MapStore{nextID: 0, store: map[uint64]Puppy{}}
+}
+
+// CreatePuppy inserts given puppy in store and returns given id to the puppy
+func (m *MapStore) CreatePuppy(puppy Puppy) uint64 {
+	m.nextID++
```

Comment:
Semantics, but imo nextID should hold the next I'd.
So sign it and increment after assignment.